internal/filter/type.go

```diff
@@ -0,0 +1,148 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package filter
+
+import (
+	"encoding/json"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"go.yaml.in/yaml/v2"
+
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+// OutputOptions handles both what information to display and how to format it.
+type OutputOptions struct {
+	infoType string // "pkgname", "dirname", "absolute"
+	format   string // "json", "yaml", ""
+}
+
+// NewOutputOptions creates a new OutputOptions from string parameters.
+func NewOutputOptions(infoType, format string) (*OutputOptions, error) {
+	cfg := &OutputOptions{
+		infoType: infoType,
+		format:   format,
+	}
+	if err := cfg.validate(); err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+func (o *OutputOptions) validate() error {
+	validInfo := []string{"pkgname", "dirname", "absolute"}
```

Comment:
Nit. As this is exposed as user options I would use `name` or `package_name` for the package name, and maybe `absolute_path` instead of `absolute`.
```suggestion
	validInfo := []string{"package_name", "dir_name", "absolute_path"}
```